src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -369,6 +380,242 @@ void CascadiaSettings::_LoadDynamicProfiles()
+        co_await resume_background();
+        const auto localExtensions = catalog.FindAllAsync().get();
+
+        std::unique_lock<std::mutex> lock{ mtx };
+        extensions = localExtensions;
+        cv.notify_all();
+    };
+
+    std::unique_lock<std::mutex> lock{ mtx };
+    lambda();
+    cv.wait(lock);
+
+    for (auto ext : extensions)
+    {
+        // Only apply the stubs if the package name is not in ignored namespaces
+        if (ignoredNamespaces.find(ext.Package().DisplayName().c_str()) == ignoredNamespaces.end())
+        {
+            // Likewise, getting the public folder from an extension is an async operation
+            // So we use another mutex and condition variable
+            std::condition_variable cv2;
+            std::mutex mtx2;
+            Windows::Storage::StorageFolder foundFolder{ nullptr };
+
+            auto lambda2 = [&]() -> winrt::fire_and_forget {
+                co_await resume_background();
+                const auto localFolder = ext.GetPublicFolderAsync().get();
+                std::unique_lock<std::mutex> lock{ mtx2 };
+                foundFolder = localFolder;
+                cv2.notify_all();
+            };
+
+            std::unique_lock<std::mutex> lock2{ mtx2 };
+            lambda2();
+            cv2.wait(lock2);
+
+            // the StorageFolder class has its own methods for obtaining the files within the folder
+            // however, all those methods are Async methods
+            // you may have noticed that we need to resort to clunky implementations for async operations
+            // (they're right above)
+            // so for now we will just take the folder path and access the files that way
+            auto path = winrt::to_string(foundFolder.Path());
+            path.append(FragmentsSubDirectory);
+            const auto jsonFiles = _AccumulateJsonFilesInDirectory(til::u8u16(path));
+
+            // Provide the package name as the source
+            _AddOrModifyProfiles(jsonFiles, ext.Package().DisplayName().c_str());
+        }
+    }
+}
+
+// Method Description:
+// - Helper function to apply json stubs in the local app data folder and the global program data folder
+// Arguments:
+// - The directory to find json files in
+// - The set of ignored namespaces
+void CascadiaSettings::_ApplyJsonStubsHelper(const std::wstring_view directory, const std::unordered_set<std::wstring>& ignoredNamespaces)
+{
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };
+
+    // The json files should be within subdirectories where the subdirectory name is the app name
+    for (auto& protoExtFolder : std::filesystem::directory_iterator(root))
+    {
+        auto folderPath = protoExtFolder.path().generic_string();
+
+        // We only want the parent folder name as the source (not the full path)
+        auto source = til::u8u16(folderPath.substr(folderPath.find_last_of("/") + 1));

```

Comment:
I feel like `std::filesystem` should have a provision for finding the pieces of the path without searching it manually as a string. Or at a minimum, it should expose a constant for "/" as the path separator without having to define it again here.

Reply:
The only thing I could find for getting _only_ the folder name (without the rest of the path) was iterating through the path elements - they do expose a path separator constant though so I've switched to that

Reply:
Can you use `protoExtFolder.filename()` or `protoExtFolder.parent_path().filename()`?

Reply:
(The "filename" of a folder is the name of the folder)